File: FrontEnd-ProjectReact/my-react-app-inventory/src/types/role.ts
import type { RolePermission } from "./rolePermission";
import type { UserAccount } from "./userAccount";

export interface Role {
  id: number;
  roleName: string;
  description?: string | null;
  recStatus?: string | null;
  rolePermission?: RolePermission[] | null;
  userAccount?: UserAccount[] | null;
}

File: Sample/Sample.Infrastructure/EntityFramework/Role.cs
using System;
using System.Collections.Generic;

namespace Sample.Infrastructure.EntityFramework
{
    public partial class Role
    {
        public Role()
        {
            RolePermission = new HashSet<RolePermission>();
            UserAccount = new HashSet<UserAccount>();
        }

        public int Id { get; set; }
        public string RoleName { get; set; } = null!;
        public string? Description { get; set; }
        public string? RecStatus { get; set; }

        public virtual ICollection<RolePermission> RolePermission { get; set; }
        public virtual ICollection<UserAccount> UserAccount { get; set; }
    }
}
